.github/workflows/build-and-test.yml

```diff
@@ -27,50 +29,96 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-      - run: scripts/ci/build-and-test.sh
-        env:
-          NODE_VERSION: ${{ matrix.node-version }}
-
-  build-and-test-windows:
-    runs-on: windows-latest
-    strategy:
-      matrix:
-        node-version: [14.x, 16.x, 18.x]
-
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v3
-        with:
-          node-version: ${{ matrix.node-version }}
-      - name: Fix node-gyp
-        run: |-
-          npm install --global node-gyp@latest
-          npm prefix -g | % {npm config set node_gyp "$_\node_modules\node-gyp\bin\node-gyp.js"}
-        shell: pwsh
       - run: bash scripts/ci/build-and-test.sh
         shell: bash
         env:
-          NODE_VERSION: ${{ matrix.node-version }}
+          SKIP_EXAMPLES: true
+      - run: scripts/ci/test-examples.sh
+        shell: bash
+        env:
+          SETUP_DIST_ONLY: true
+
+      ## require jest-pact and mocha-pact to update peer deps
+      # - name: example_mocha
+      #   run: cd examples/mocha && npm install --ignore-scripts && npm test
+      # - name: example_jest
+      #   run: cd examples/jest && npm install --ignore-scripts && npm test
+
+
+      #  passing
+      - name: example_e2e
+        run: cd examples/e2e && npm install --ignore-scripts && npm test
+      - name: example_graphql
+        run: cd examples/graphql && npm install --ignore-scripts && npm test
+      - name: example_messages
+        run: cd examples/messages && npm install --ignore-scripts && npm test
+      - name: example_serverless
+        run: cd examples/serverless && npm install --ignore-scripts && npm test
+      - name: example_typescript
+        run: cd examples/typescript && npm install --ignore-scripts && npm test
+      - name: example_v3_e2e
+        run: cd examples/v3/e2e && npm install --ignore-scripts && npm test
+      - name: example_v3_provider-state-injected
+        run: cd examples/v3/provider-state-injected && npm install --ignore-scripts && npm test
+      - name: example_v3_run-specific-verifications
+        run: cd examples/v3/run-specific-verifications && npm install --ignore-scripts && npm test
+      - name: example_v3_todo-consumer
+        run: cd examples/v3/todo-consumer && npm install --ignore-scripts && npm test
+      - name: example_v3_typescript
+        run: cd examples/v3/typescript && npm install --ignore-scripts && npm test
+      - name: example_v4_plugins
+        run: cd examples/v4/plugins && npm install --ignore-scripts && npm test          
+      # - name: Upload dist folder
+      #   if: runner.os == 'Linux'
+      #   uses: actions/upload-artifact@v3
+      #   with:
+      #     path: dist
 
-  # Failures not reproducible locally on an M1 Mac
-  # build-and-test-macos:
-  #   runs-on: macos-latest
+  # examples:
```

Comment:
I tried to see if I could package up the dist folder and distribute it to each of the examples in separate builds but I couldn't get what I wanted quickly enough and I was already several yaks deep

Reply:
One herd at a time my friend! Let's pick this up in a subsequent piece of work.